src/algorithms/optimized_shooting.jl

```diff
@@ -0,0 +1,116 @@
+export periodic_orbit, periodic_orbits, OptimizedShooting
+
+using LeastSquaresOptim: optimize, LevenbergMarquardt
+
+
+"""
+    OptimizedShooting(; kwargs...)
+
+A shooting method [Dednam2014](@cite) combined with Levenberg-Marquardt optimization 
+to find periodic orbits of continuous systems.
+
+## Keyword arguments
+- `Δt::Float64=1e-6`: explicit ODE solver time step. This should correspond to the time step used in the ODE solver specified in the `CoupledODEs` object.
+- `p::Int64=2`: `p*dimension(ds)` is the number of points in the residual `R`.
```

Comment:
Don't use `p` here. Use something else. `p` is a reserved symbol throghout DynamicalSystems.jl used to represent system parameters.